File: design-patterns/java/builder/Pizza.java
class Pizza {
    private String dough = "";
    private String sauce = "";
    private String topping = "";

    public void setDough(String dough)     { this.dough = dough; }
    public void setSauce(String sauce)     { this.sauce = sauce; }
    public void setTopping(String topping) { this.topping = topping; }

    public String toString() {
        return "Pizza with " + dough + " dough, " + sauce + " sauce, and " + topping + " topping.";
    }
}

File: design-patterns/go/builder/pizza/types/pizza.go
package pizza

import "fmt"

type Pizza struct {
	Dough    string
	Sauce    string
	Toppings []string
}

func NewPizza() *Pizza {
	return &Pizza{
		Toppings: make([]string, 0),
	}
}

func (p Pizza) Description() string {
	s := "Pizza has %s dough, %s sauce, %s topping\n"
	return fmt.Sprintf(s, p.Dough, p.Sauce, p.Toppings)
}
